protocol/chainlib/jsonRPC.go

```diff
@@ -61,7 +61,12 @@ func (apip *JsonRPCChainParser) getSupportedApi(name, connectionType string, int
 
 func (apip *JsonRPCChainParser) CraftMessage(parsing *spectypes.ParseDirective, connectionType string, craftData *CraftData, metadata []pairingtypes.Metadata) (ChainMessageForSend, error) {
 	if craftData != nil {
-		chainMessage, err := apip.ParseMsg("", craftData.Data, craftData.ConnectionType, metadata, extensionslib.ExtensionInfo{LatestBlock: 0})
+		path := craftData.Path
+		if craftData.InternalPath != "" {
+			path = craftData.InternalPath
```

Comment:
we overwrite path here is that intended or we need to use path+/+internal_path?

Reply:
Yes, this is intended. You can look at the use of this field here:
https://github.com/lavanet/lava/blob/dcffaf73574f46cec25b511a02b12a2188e0ffa1/protocol/chainlib/jsonRPC.go#L117C4-L117C37

Reply:
Yes, this is intended. You can look at the use of this field here:

https://github.com/lavanet/lava/blob/6591537984b4e0e305ab4c781b7a6a1fad5ce5ce/protocol/chainlib/jsonRPC.go#L112